### PR TITLE
LPS-194344 Scrape the spritemap and add RequiredMark for required inputs

### PR DESCRIPTION
**Pull request by kresimir-coko**
Continuation of https://github.com/brianchandotcom/liferay-portal/pull/139849
A proper variant of https://github.com/liferay-frontend/liferay-portal/pull/3622

Reason for a new PR is that a lot of changes happened in the meanwhile (mostly by @thektan), and conflict resolution.

In this PR I am doing the following:
- passing `spritemap` from the backend to our `Actions` component
- consuming the `spritemap` in a `ClaySelect` that offers the scraped icons
- adjusting the icon preview that updates as the user is changing the icon in the `ClaySelect`
- importing and using the `RequiredMark` component to mark inputs as required

### CODE REVIEW

**Comment by mateusralv** [comment]
Hey @kresimir-coko the failures are not related.

[Testray comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1693855304513&testrayRunIdA=2418030551&testrayRunIdB=2418179305)

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx [hunk 1]
```diff
@@ -68,19 +70,53 @@ const TYPES = [
...
 	const [
 		confirmationMessageTranslations,
 		setConfirmationMessageTranslations,
 	] = useState({});
 	const [iconSymbol, setIconSymbol] = useState('bolt');
 	const [labelTranslations, setLabelTranslations] = useState({});
 
+	useEffect(() => {
+		const getIcons = async () => {
+			const response = await fetch(spritemap);
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx:88]: First time I see spritemap used this way :open_mouth: Nice!

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx [hunk 2]
```diff
@@ -68,19 +70,53 @@ const TYPES = [
...
+
+			if (responseText.length) {
+				const XMLString = await new window.DOMParser().parseFromString(
+					responseText,
+					'text/xml'
+				);
+
+				const availableIconSymbolElements = XMLString.querySelectorAll(
+					'symbol'
+				);
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx:100]: Interesting way of parsing. Pure string parsing, with a regex, would probably have better performance, but I think a small performance hit is worth for the simplification in code. :+1: 

Is `await` needed?

nits:
- `XMLString` should normally be lowercase, `xmlString`. But in this case, the variable does not contain a string value. It should be something like `spritemapDocument`.
- `window` is not necessary
- `availableIconSymbolElements` can be just `symbolElements`

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx [hunk 3]
```diff
@@ -254,14 +287,16 @@ const Actions = () => {
...
+												<RequiredMark />
 											</label>
 
 											<ClaySelectWithOption
 												defaultValue="link"
 												id="actionTypeSelect"
 												options={TYPES}
 												placeholder={Liferay.Language.get(
-													'select-an-option'
+													'please-select-an-option'
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx:299]: :heart:

**Review by markocikos — COMMENTED** [review]
@kresimir-coko please see comments inline.

This will have conflicts with https://github.com/liferay-frontend/liferay-portal/pull/3639, but that's fine. This one is likely to be the first one merged.

**Comment by kresimir-coko** [comment]
I've rebased and force-pushed the changes, nothing significant but it covers all your comments @markocikos I think this should be forwardable, thoughts?

**Review by markocikos — APPROVED** [review]
LGTM! :sparkles:

**Comment by kresimir-coko** [comment]
ci:forward

**Comment by kresimir-coko** [comment]
ci:forward

**Comment by kresimir-coko** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:forward

**Comment by kresimir-coko** [comment]
ci:forward:force

**Comment by markocikos** [comment]
ci:forward:force

**Comment by markocikos** [comment]
manual forward: https://github.com/brianchandotcom/liferay-portal/pull/140463

**Pull request closed without merging**